### DISCUSSION TITLE
Geodesic flow

### DISCUSSION THREAD

**Discussion by joshp112358** [JuliaManifolds/Manifolds.jl#554] [Q&A]
I want to study dynamical properties of geodesic flow on my computer. But I cannot figure out how to even start! I am brand new to Julia. If someone is so kind, could someone show me the minimal working example to get a geodesic flow on the sphere and torus working, with the standard Riemannian metric and plot a time series with the surface? I want to be able to make something like this picture https://mathoverflow.net/questions/81622/probing-a-manifold-with-geodesics except for various initiial conditions.

I would also like to know if/how to change Riemannian metrics.

**Comment by kellertuer** [comment]
Hej,
sure we can help you get started.
I am just not 100% sure where we can best start to help you, can you maybe post a beginning what you tried or a reference to an example you plan to implement or something like that?
For example I do not directly have an idea what the time series with the embedded manifold would look like.

**Reply by joshp112358** [reply]
Hi, I just edited it to be a bit more clear. I'd like to be able to make pictures like http://www.rdrop.com/~half/math/torus/illustrating.html except without the excessive handwork. Or, for example, if I could write down this surface, https://sites.math.northwestern.edu/~burns/ , whether I could simulate the geodesic flow on it. But I would take just the standard S1 x S1 for now. Then I could play around.

**Reply by kellertuer** [reply]
Hm, I fear I personally would need more mathematics here.

A first remark on the surface – we do not do surfaces in this package (much) besides the Torus, the Sphere and the hyperbolic manifold (in the hyperboloid model).

For the first link here, I do not understand exactly, what they are doing mathematically, but we do have charts, which is a way of having these 2D maps of the torus or the sphere – but for the sphere only locally of course.
For the second link … we do not cover support for surfaces, and I fear that would be a completely new package to cover that. One difference for us here is that we usually have closed form solutions for the exponential and the logarithmic map and hence for the geodesic, which is not the case on the surfaces, where one usually has to solve a second order ODE.

So could you maybe provide a little bit more mathematical background?

**Reply by mateuszbaran** [reply]
We actually do have visualizations of geodesics on a torus: https://github.com/JuliaManifolds/Manifolds.jl/blob/master/tutorials/working-in-charts.jl . You can experiment with it using sliders or by changing parameters.

I was actually thinking about extending it to arbitrary parametric surfaces. @joshp112358 , if you would like to work on such visualizations I could help you with working out the implementation details. For a start let me know if what's in that torus tutorial is what you have in mind. Feel free to write to me on Julia Slack or Zulip if you want a real-time conversation :slightly_smiling_face: .

**Reply by joshp112358** [reply]
Thanks! I just arose (East Coast US Here). I will see if I can figure things out... but yes I am interested. I'll ping you in Slack.
